packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx

```diff
@@ -96,17 +145,59 @@ const CompleteResetPassword = (_: RouteComponentProps) => {
   }, [navigate]);
 
   const onSubmit = useCallback(
-    async ({ newPassword }: FormData) => {
+    async ({ newPassword, token, code, email }: SubmitData) => {
       try {
-        await account.completeResetPassword(token!, code!, email!, newPassword);
+        // TODO: do we no longer need emailToHashWith?
+        await account.completeResetPassword(token, code, email, newPassword);
```

Comment:
The `emailToHashWith` (terrible name), should be used if it is avaible instead of email. Ref: https://github.com/mozilla/fxa/blob/687fedd36f6c9adfebe87e1ba02eb246647d0665/packages/fxa-content-server/app/scripts/lib/fxa-client.js#L629

Reply:
@vbudhram Ahhh I see. Is this supposed to _always_ be provided, or only when the user's changed their primary email? I think I'm actually going to tweak this in a follow up I'll throw into next sprint - this requires a couple (still easy) changes to `useLinkStatus` and tests around this. This doesn't change the functionality in my PR - we were doing this [here](https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx#L101). 😬